src/interfaces.ts

```diff
@@ -5,118 +5,193 @@ export interface Config {
 	fetch?: Fetch
 }
 
-export interface ModelParameters {
-	mirostat: number
-	mirostat_eta: number
-	mirostat_tau: number
-	num_ctx: number
-	num_gqa: number
-	num_thread: number
-	repeat_last_n: number
-	repeat_penalty: number
-	temperature: number
-	stop: string
-	tfs_z: number
-	top_k: number
-	top_p: number
-}
-
-export interface GenerateOptions {
-	parameters: Partial<ModelParameters>
-	context: number[]
-	template: string
-	system: string
+// request types
+
+export interface Options {
+    numa: boolean;
```

Comment:
Should we keep these consistent with the REST api and python library?

E.g. `num_ctx`

Reply:
Where these are properties of an interface I'd rather follow the language convention personally. In Python that is snake-case (matches our api), but in JavaScript and Go that is camel-case.